### ISSUE TITLE
result of mixedlm  regressions do not work with summary_col

### ISSUE THREAD

**Issue by KostasTok** [statsmodels/statsmodels#3320]
The following will return an error:
```python
import statsmodels.formula.api as smf
import pandas as pd

ids = [1,1,1,1,1,2,2,2,2,3,3,3,3,3,3]
x = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15]
y = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15]

d = {'Y':y,'X':x,'IDS':ids}
d = pd.DataFrame(d)

results1 = smf.mixedlm('Y ~ X', d, groups=d ['IDS']).fit()
results2 = smf.mixedlm('X ~ Y', d, groups=d ['IDS']).fit()
    
out = summary_col([results1,results2],stars=True)
```

I think that the problem is created from statsmodels/iolib/summary2.py lines 349-52. As a quick fix I am working with the following code for now.
```python
if not xname:
    if len(data) != len(results.model.exog_names):
        data.index = np.append(results.model.exog_names,'empty')
    else:
        data.index = results.model.exog_names
else:
    data.index = xname
```

**Comment by josef-pkt** [comment]
@KostasTok Thanks for reporting.

Can you try with using `results.model.data.param_names`? (at least in an ols example, it's a `model.data` but not a `model` attribute.)
AFAIR, we switched to using `param_names` to handle cases with len(params) > k_exog, but it looks like it wasn't changed in summary2 or summary_col.

**Comment by KostasTok** [comment]
@josef-pkt 

Works perfectly. Thank you.

**Comment by josef-pkt** [comment]
To make sure: Was switching to param_names the only change that you needed to make?

Do you want to make a PR for the change?
There might be a place to add a smoke test (just call the function to see that it doesn't raise an exception) to an existing MixedLM test.

**Comment by KostasTok** [comment]
Yes, just  switching to param_names was enough, and I tried it on various datasets and formulas.

I will also make the PR. Thanks.

**Comment by josef-pkt** [comment]
Thanks, we keep this issue open until a fix is merged, or it might get lost.

**Comment by josef-pkt** [comment]
fixed in #4505
